src/core/lib/surface/validate_metadata.h

```diff
@@ -24,7 +24,18 @@
 #include <grpc/slice.h>
 #include "src/core/lib/iomgr/error.h"
 
-grpc_error* grpc_validate_header_key_is_legal(grpc_slice slice);
-grpc_error* grpc_validate_header_nonbin_value_is_legal(grpc_slice slice);
+grpc_error* grpc_validate_header_key_is_legal(const grpc_slice& slice);
+grpc_error* grpc_validate_header_nonbin_value_is_legal(const grpc_slice& slice);
+
+int grpc_is_binary_header_internal(const grpc_slice& slice);
```

Comment:
validate_metadata.h does not seem to be the right place for this

Reply:
Where would you recommend? The validate_metadata.cc file has the concrete implementations.

Reply:
Sorry for the late reply, there doesn't seem to be an existing header file where this can be placed. The only other solution would be to create a new header file altogether, but as it stands, I'm fine with it being here.